containerWithMostWater.js

```diff
@@ -0,0 +1,23 @@
+// This file contains solution to the following problem:
+// https://leetcode.com/problems/container-with-most-water/
+
+const maxArea = (height) => {
+  let pointer1 = 0;
+  let pointer2 = height.length - 1;
+  let sum = Math.min.apply(null, height);
```

Comment:
What's this for ? Can't we just do: `let sum = height` ? If can't - then leave explaining comment for "why we need to take 'min' here".

Reply:
No, we cannot use let sum = height, because the area is restricted by the minimum number. In my algorithm I compute the biggest area and compare it with the smaller area. For example, if we take [8, 1]  the biggest area is 1, not 8.

Reply:
I guess what bothers me about this is comparing `null` against a `number`(which is what happens here because you gotta be able to tell which is higher/lower to find **min** value),

even though in JS there are all kinds of different **implicit** type conversions - you should never rely on them (because for me as a reader it's hard to tell what will happen since I never actually ran this code).

This is why I suggest to rewrite it. I haven't looked into exact semantics of how it **should** work (so my suggestion above is just probably a wrong guess), but you can find a different way to do this I guess.